addons/mail/static/src/models/user_notification_manager.js

```diff
@@ -0,0 +1,149 @@
+                }
+                try {
+                    this._sendNativeNotification(title, message);
+                } catch (error) {
+                    // Notification without Serviceworker in Chrome Android doesn't works anymore
+                    // So we fallback to the notification service in this case
+                    // https://bugs.chromium.org/p/chromium/issues/detail?id=481856
+                    if (error.message.includes('ServiceWorkerRegistration')) {
+                        this._sendOdooNotification(message, { title, type });
+                    } else {
+                        throw error;
+                    }
+                }
+            } else {
+                this._sendOdooNotification(message, { title, type });
+            }
+        },
+        /**
+         * @private
+         * @returns {HTMLAudioElement}
+         */
+        _computeAudio() {
+            const audioElement = new Audio();
+            audioElement.src = audioElement.canPlayType("audio/ogg; codecs=vorbis")
+                ? url('/mail/static/src/audio/ting.ogg')
+                : url('mail/static/src/audio/ting.mp3');
+            return audioElement;
+        },
+        /**
+         * Determines whether or not sending native notification is
+         * allowed.
+         *
+         * @private
+         * @returns {boolean}
+         */
+        _computeCanSendNativeNotification() {
+            return (
+                this.messaging.browser.Notification &&
+                this.messaging.browser.Notification.permission === 'granted'
+            );
+        },
+        /**
+         * Method to be called when the users click on a notification.
+         *
+         * @private
+         */
+        _onClickNotification() {
+            window.focus();
+            this.notification.close();
+        },
+        /**
+         * Play the notification sound.
+         */
+        async _playAudio() {
+            if (this.canPlayAudio) {
+                try {
+                    await this.audio.play();
+                } catch {
+                    // Ignore errors due to the user not having interracted
+                    // with the page before playing the sound.
+                }
+            }
+        },
+        /**
+         * Send a native notification.
+         *
+         * @param {string} title
+         * @param {string} message
+         */
+        _sendNativeNotification(title, message) {
+            const notification = new Notification(
+                // The native Notification API works with plain text and not HTML
+                // unescaping is safe because done only at the **last** step
+                _.unescape(title),
+                {
+                    body: _.unescape(message),
+                    icon: this.icon,
+                }
+            );
+            notification.addEventListener('click', this._onClickNotification);
+            this.update({ notification });
```

Comment:
What if there are multiple open notifications? Won't this be an issue?

Reply:
Not needed after all, the event passed to the `onclick` handler holds a reference to the Notification itself. Field removed.